tools/pmd_test.bzl

```diff
@@ -7,38 +7,71 @@ def _impl(ctx):
         fail("Expecting a single java library")
 
     src_jar = lib.source_jars[0]
-    full_compile_jars = ":".join([f.path for f in lib.full_compile_jars.to_list()])
+    jar_deps = lib.full_compile_jars
+    full_compile_jars = ":".join([f.short_path for f in jar_deps.to_list()])
```

Comment:
Q: was it obvious to you how to choose between `path`, `short_path`, etc?

Reply:
It was not - my first rules were all .path and many runs of --sandbox_debug --verbose_failures trying to figure out what was going on. The docs do explicitly mention short_path is correct for runfiles, but I hadn't gotten that far, having seen .path and running with it.